File: Hospital.Server/Entities/Response/DoctorTaskResponse.cs
namespace Hospital.Server.Entities.Response
{
    public class DoctorTaskResponse
    {
        public long Id { get; set; }
        public long DoctorId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string DueDate { get; set; } = string.Empty;
        public bool IsCompleted { get; set; }
        public int Priority { get; set; }
        public int State { get; set; }
        public string CreatedAt { get; set; } = string.Empty;
        public long CreatedBy { get; set; }
        public long? UpdatedBy { get; set; }
        public string? UpdatedAt { get; set; }
        public string? DoctorName { get; set; }
    }
}

File: hospital.client/src/types/DoctorTaskResponse.ts
export interface DoctorTaskResponse {
  id: number;
  doctorId: number;
  title: string;
  description?: string | null;
  dueDate: string;
  isCompleted: boolean;
  priority: number;
  state: number;
  createdAt: string;
  createdBy: number;
  updatedBy?: number | null;
  updatedAt?: string | null;
  doctorName?: string | null;
}

export interface DoctorTaskRequest {
  id?: number | null;
  doctorId?: number | null;
  title?: string | null;
  description?: string | null;
  dueDate?: string | null;
  isCompleted?: boolean | null;
  priority?: number | null;
  state?: number | null;
  createdBy?: number | null;
  updatedBy?: number | null;
}

/**
 * Maps numeric Priority values to Spanish labels and TailwindCSS colors.
 * 0=Baja, 1=Normal, 2=Alta
 */
export const PriorityLabels: Record<number, { label: string; color: string }> = {
  0: { label: "Baja", color: "bg-gray-100 text-gray-800" },
  1: { label: "Normal", color: "bg-blue-100 text-blue-800" },
  2: { label: "Alta", color: "bg-red-100 text-red-800" },
};
